contracts/Marketplace1155.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Counters.sol";
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol";
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/security/ReentrancyGuard.sol";
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/utils/ERC1155Receiver.sol";
+
+contract marketPlace1155 is ReentrancyGuard, ERC1155Receiver {
+    using Counters for Counters.Counter;
+    Counters.Counter private _itemIds;
+    Counters.Counter private _itemsSold;
+
+    address public owner;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    struct MarketItem {
+        uint256 itemId;
+        address nftContract;
+        uint256 tokenIds; // togli s finale
+        address payable seller;
+        address payable owner;
+        uint256 price;
+        bool sold; //serve?
+    }
+
+    mapping(uint256 => MarketItem) private idToMarketItem; // a cosa mi serve la mappa. Sto rimappando i dati di 1155. Se non aggiunge informazione a cosa serve?
+
+    event MarketItemCreated(
+        uint256 indexed itemId,
+        address indexed nftContract,
+        uint256 indexed tokenIds,
+        address seller,
+        address owner,
+        uint256 price,
+        bool sold
+    );
+    // serve il bool sold? quando lo creo puo' essere gia' venduto?
+
+    event MarketItemSold(uint256 indexed itemId, address owner);
+
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155Received.selector;
+    }
+
+    function onERC1155BatchReceived(
+        address,
+        address,
+        uint256[] memory,
+        uint256[] memory,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155BatchReceived.selector;
+    }
+
+    function createMarketItem(
+        address nftContract,
+        uint256[] memory tokenIds,
+        uint256 price,
+        uint256[] memory amounts // se e' un nft amounts non dovrebbe essere sempre 1? siccome immagino siano tracked dai tokenIds questo mi sembra superfluo
+    ) public payable nonReentrant {
+        require(price > 0, "Price must be greater than 0"); // Non vogliamo controllare che tokenIds sia lungo quanto amounts?
+
```

Comment:
Should we check also that `tokenIds` has the same length of amounts? (If amounts can have values greater than 1)